pkg/loop/server.go

```diff
@@ -44,10 +51,13 @@ func MustNewStartedServer(loggerName string) *Server {
 
 // Server holds common plugin server fields.
 type Server struct {
-	GRPCOpts   GRPCOpts
-	Logger     logger.SugaredLogger
-	promServer *PromServer
-	checker    *services.HealthChecker
+	GRPCOpts        GRPCOpts
+	Logger          logger.SugaredLogger
+	db              *sqlx.DB
+	dbStatsReporter *pg.StatsReporter
+	DataSource      sqlutil.DataSource
```

Comment:
what do we intend `db` to be used for vs `DataSource`? a comment may be helpful

Reply:
We hang to `db` in order to call `Close`. We could store `dbCloseFn func() error = db.Close` instead, but it is not exported so seems like nbd :shrug: